### PR TITLE
HDS-1589 hds-demo preview workflows

### PR DESCRIPTION
**Pull request by terovirtanen**
## Description

Workflow files to update hds-demo -repository

[HDS-1589](https://helsinkisolutionoffice.atlassian.net/browse/HDS-1589)

## Related Issue

Closes #

## Add to changelog
- [ ] Added needed line to changelog 

[HDS-1589]: https://helsinkisolutionoffice.atlassian.net/browse/HDS-1589?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mrTuomoK** [comment]
@terovirtanen could you squash all the commits as one?

**Comment by terovirtanen** [comment]
> @terovirtanen could you squash all the commits as one?

I planned to squash on merge. Is it ok?

**Comment by mrTuomoK** [comment]
> > @terovirtanen could you squash all the commits as one?
> 
> I planned to squash on merge. Is it ok?

Ok that works as well 👍

File: .github/workflows/hds-demo-preview.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
+name: hds-demo-preview
+
+on:
+  pull_request:
+    branches:
+      - development
```
Comment (mrTuomoK) [inline .github/workflows/hds-demo-preview.yml:6]: I'm just wondering that should we also make these demos for the PR's targeting release-x.x.x -branches?

Comment (terovirtanen) [reply .github/workflows/hds-demo-preview.yml:6]: added 'release-*'

File: .github/workflows/hds-demo-preview.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      - name: build site package
+        run: yarn build --prefix-paths
+        working-directory: ./site
+
+      # Build core storybook
+      - name: build core storybook
+        run: yarn build-storybook
+        working-directory: ./packages/core
+
+      - name: move core storybook under site
```
Comment (mrTuomoK) [inline .github/workflows/hds-demo-preview.yml:88]: should core storybook go under ./site/public/storybook/core?

Comment (terovirtanen) [reply .github/workflows/hds-demo-preview.yml:88]: This is copied from gh-pages.yml. 

This is how it has done on 
https://hds.hel.fi/storybook/core

Comment (mrTuomoK) [reply .github/workflows/hds-demo-preview.yml:88]: ok maybe I just missed something myself :D Monday morning :D

**Review by mrTuomoK — APPROVED** [review]
ok, good work 👍

**Pull request merged**